app/vehicles/add-vehicle.tsx

```diff
@@ -1,39 +1,53 @@
 import { Link, router, Stack } from 'expo-router'
 import { StatusBar } from 'expo-status-bar'
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 
 import TextInput from '@/components/inputs/TextInput'
 import Button from '@/components/shared/Button'
 import Field from '@/components/shared/Field'
+import Modal, { ModalContentWithActions } from '@/components/shared/Modal'
 import ScreenContent from '@/components/shared/ScreenContent'
 import ScreenView from '@/components/shared/ScreenView'
+import Surface from '@/components/shared/Surface'
 import Typography from '@/components/shared/Typography'
+import { useModal } from '@/hooks/useModal'
 import { useTranslation } from '@/hooks/useTranslation'
-import { useVehiclesStorage, Vehicle } from '@/hooks/useVehiclesStorage'
+import { useVehicles } from '@/hooks/useVehicles'
+import { Vehicle } from '@/hooks/useVehiclesStorage'
+import { isStandardFormat, sanitizeLicencePlate } from '@/utils/licencePlate'
 
-// TODO make sure that modal is what we want and that it works correctly
 const AddVehicleScreen = () => {
   const t = useTranslation('VehiclesScreen')
+  const tAddModal = useTranslation('VehiclesScreen.addVehicleConfirmModal')
```

Comment:
not sure if it helps much to structure the translation files and load only parts - in RN, you always have all of them bundled and "downloaded" already anyway. I'd just use translation with 'root namespace and write out the entire path each time the translation is used, making it easy to navigate and change.

Just an opinion, this way is ok as well. Also, if this is done for different reason I'm missing carry on (and let me know).

Reply:
You are right, I wanted to "prepare" it for splitting into separate components, but it's just confusing and hard to read. I'll keep it one level deep in `VehiclesScreen` - that makes sense for me to keep translations grouped "by component" (or related components).

Reply:
I'll use only one `t` function and write longer paths. :)